### PR TITLE
Update samples for C#

### PR DESCRIPTION
**Pull request by drwill-ms**

### CODE REVIEW

File: sdks/csharp/model_operations.cs [hunk 1]
```diff
@@ -2,10 +2,8 @@
 // <CreateModel>
 // 'client' is an instance of DigitalTwinsClient
 // Read model file into string (not part of SDK)
-using var modelStreamReader = new StreamReader("MyModelFile.json");
-string dtdl = modelStreamReader.ReadToEnd();
-string[] dtdls = new string[] { dtdl };
-client.CreateModels(dtdls);
+string dtdl = new File.ReadAllText("MyModelFile.json");
```
Comment (drwill-ms) [inline sdks/csharp/model_operations.cs:5]: File.ReadAllText is fewer lines than using a StreamReader, requires no disposing.

File: sdks/csharp/model_operations.cs [hunk 2]
```diff
@@ -2,10 +2,8 @@
...
+await client.CreateModelsAsync(new[] { dtdl });
```
Comment (drwill-ms) [inline sdks/csharp/model_operations.cs:6]: Stick with async client methods

File: sdks/csharp/model_operations.cs [hunk 3]
```diff
@@ -16,29 +14,29 @@
...
+await client.CreateModelsAsync(dtdlModels);
 // </CreateModels_multi>
 
 // ------------------ GET MODELS ---------------------
 // <GetModels>
 // 'client' is a valid DigitalTwinsClient object
 
 // Get a single model, metadata and data
-DigitalTwinsModelData md1 = client.GetModel(id);
+Response<DigitalTwinsModelData> md1 = await client.GetModelAsync(id);
```
Comment (drwill-ms) [inline sdks/csharp/model_operations.cs:28]: return type was wrong

File: sdks/csharp/model_operations.cs [hunk 4]
```diff
@@ -16,29 +14,29 @@
...
+DigitalTwinsModelData model1 = md1.Value;
```
Comment (drwill-ms) [inline sdks/csharp/model_operations.cs:29]: Illustrate how to get the model

File: sdks/csharp/twin_operations_other.cs [hunk 5]
```diff
@@ -1,27 +1,53 @@
 // ------------------ CREATE TWIN: NO HELPER ---------------------
 // <CreateTwin_noHelper>
-// Define the model type for the twin to be created
-using System.Net.Mime;
```
Comment (drwill-ms) [inline sdks/csharp/twin_operations_other.cs:4]: I was drunk when I added this, apparently

File: sdks/csharp/twin_operations_other.cs [hunk 6]
```diff
@@ -1,27 +1,53 @@
...
+// Define a custom model type for the twin to be created
+class CustomDigitalTwin
```
Comment (drwill-ms) [inline sdks/csharp/twin_operations_other.cs:4]: This defines a custom class to deserialize my digital twin

File: sdks/csharp/twin_operations_other.cs [hunk 7]
```diff
@@ -1,27 +1,53 @@
...
-        { "Humidity", 50.0},
-    },
+    Temperature = 25.0,
+    Humidity = 50.0,
 };
-//Create the twin
-client.CreateOrReplaceDigitalTwinAsync(twinID, twinInit);
+// Create the twin
+Response<BasicDigitalTwin> response = await client.CreateOrReplaceDigitalTwinAsync(twinId, myTwin);
+Console.WriteLine($"Temperature last updated on {response.Value.Metadata.Temperature.LastUpdatedOn}")
```
Comment (drwill-ms) [inline sdks/csharp/twin_operations_other.cs:43]: Illustrate how to use info in metadata

File: sdks/csharp/model_operations.cs (hunk 3 above)
Comment (azabbasi) [reply sdks/csharp/model_operations.cs:28]: `ConfigureAwait(false)`? and elsewhere

**Review by azabbasi — APPROVED** [review]

**Pull request merged**